### PR TITLE
Update pallet teerex to frame v2

### PR DESCRIPTION
**Pull request by haerdib**
Metadata of pallet did not change at all.

### CODE REVIEW

File: teerex/src/lib.rs [hunk 1]
```diff
@@ -40,72 +39,130 @@ use ias_verify::SgxBuildMode;
...
+	#[pallet::genesis_build]
+	impl<T: Config> GenesisBuild<T> for GenesisConfig {
+		fn build(&self) {
+			{
+				let data = &self.allow_sgx_debug_mode;
+				let v: &bool = data;
+				<AllowSGXDebugMode<T> as frame_support::storage::StorageValue<bool>>::put::<&bool>(
+					v,
+				);
+			}
```
Comment (clangenb) [inline teerex/src/lib.rs:132]: This one is unnecessarily verbose due to the migration template, you can simplify this to a one liner:

```
<AllowSGXDebugMode<T>>::put(&self.allow_sgx_debug_mode);

```

**Review by clangenb — COMMENTED** [review]
Looks good to me, one style comment.

File: teerex/src/lib.rs (hunk 1 above)
Comment (haerdib) [reply teerex/src/lib.rs]: Good catch, updated accordingly

**Review by clangenb — APPROVED** [review]
Coolio

**Pull request merged**